debezium-core/src/test/java/io/debezium/data/VerifyRecord.java

```diff
@@ -250,6 +250,77 @@ public static void isValidTombstone(SourceRecord record, String pkField, int pk)
         isValidTombstone(record);
     }
 
+    /**
+     * Verify that the given {@link SourceRecord} has the appropriate source query value.
+     *
+     * @param record the source record; may not be null
+     * @param query the expected sql query value.
+     */
+    public static void hasValidSourceQuery(final SourceRecord record, final String query) {
+        assertValueField(record, "source/query", query);
+    }
+
+    /**
+     * Verify that the given {@link SourceRecord} has no source query value.
+     *
+     * @param record the source record; may not be null
+     */
+    public static void hasNoSourceQuery(final SourceRecord record) {
+        // Abstracted out so if we change from having a null value in the query field to
+        // not including the query field entirely, we can easily update the assertion in one place.
+        hasValidSourceQuery(record, null);
+    }
+
+    /**
+     * Verify the given {@link SourceRecord} has the expected value in the given fieldPath.
+     *
+     * @param record the source record; may not be null
+     * @param fieldPath the field path to validate, separated by '/'
+     * @param expectedValue the expected value in the source records field path.
+     */
+    public static void assertValueField(SourceRecord record, String fieldPath, Object expectedValue) {
+        Object value = record.value();
+        String[] fieldNames = fieldPath.split("/");
+        String pathSoFar = null;
+        for (int i=0; i!=fieldNames.length; ++i) {
+            String fieldName = fieldNames[i];
+            if (value instanceof Struct) {
+                value = ((Struct)value).get(fieldName);
+            } else {
```

Comment:
Minor nit on style: can you move the "else {" to the next line, i.e. there should be no code after closing braces. I know it's not done consistently in the code base, but with Jiri we recently agreed on using that style going forward in any new or changed code. Thanks!